src/Core/FormatFactorySettings.h

```diff
@@ -953,6 +953,9 @@ Parquet format version for output format. Supported versions: 1.0, 2.4, 2.6 and
 )", 0) \
     DECLARE(ParquetCompression, output_format_parquet_compression_method, "zstd", R"(
 Compression method for Parquet output format. Supported codecs: snappy, lz4, brotli, zstd, gzip, none (uncompressed)
+)", 0)                                          \
+    DECLARE(UInt64, output_format_parquet_compression_level, 3, R"(
```

Comment:
There already exists a generic setting for compression level, should it be re-used? https://github.com/ClickHouse/ClickHouse/blob/6d2e179eef5e94844f5fbe56dbbc7f7ce9c50a45/src/Core/Settings.cpp#L1222.

The only caveat is that it is not declared inside `FormatFactorySettings`, but rather on `Settings.cpp`. I am not exactly sure what that really means.